src/daria/corrections/shape/curvature.py

```diff
@@ -387,18 +388,46 @@ def compute_stretch(self, img: Optional[np.ndarray] = None, **kwargs):
         pt_src = kwargs.pop("point_source", [Ny, Nx])
         pt_dst = kwargs.pop("point_destination", [Ny, Nx])
         stretch_center = kwargs.pop("stretch_center", [round(Ny / 2), round(Nx / 2)])
-        # TODO ADD WARNING!!
+
         # Update the offset to the center
         horizontal_stretch_center_offset = stretch_center[0] - round(Nx / 2)
         vertical_stretch_center_offset = stretch_center[1] - round(Ny / 2)
 
-        # Compute the tuning parameter as explained in the notes
-        horizontal_stretch = -(pt_dst[0] - pt_src[0]) / (
-            (pt_src[0] - stretch_center[0]) * pt_src[0] * (Nx - pt_src[0])
-        )
-        vertical_stretch = -(pt_dst[1] - pt_src[1]) / (
-            (pt_src[1] - stretch_center[1]) * pt_src[1] * (Ny - pt_src[1])
-        )
+        # Check whether zero horizontal stretch should be applied
+        if (pt_dst[0] - pt_src[0]) == 0:
+            horizontal_stretch = 0
+        # Check whether point is chosen too close to the center
+        # (within 10% of total pixels), and make warning
+        elif abs(pt_src[0] - stretch_center[0]) < round(0.1 * Nx):
+            horizontal_stretch = 0
+            warn(
+                "point_source chosen too close to stretch center for correction"
+                " in horizontal direction. Please choose a value towards the corners."
+                " Horizontal stretch is set to zero now."
```

Comment:
I think it is best to mention the 10% in the warning. It is actually important that the value is also not too close to the corners, as then `Nx - pt_src[0]` or `pt_src[0]' also approache 0. So in principle, one should also check for these in the same condition. As we set up the stretch correction, the best would be to choose some value in the center of the four respective quadrants if this makes sense. So my suggestion would be to amend this to the check, as well as be more precise in the warning, to allow the user to make an adjustment without the need to guess.

Reply:
I made a suggestion. Perhaps you could take a look before we merge. I chose to split it into three parts so that the warning tells the user which "mistake" is being made. Moreover, I adjusted it to 5% of the image's pixels as 10% seemed a bit much. What do you think?